### PR TITLE
fix(user.ip.restriction): prevent removing last ip on deny default rule

### PR DESCRIPTION
**Pull request by antleblanc**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | #4052
| License          | BSD 3-Clause

### :bug: Bug Fixes

5ed1eed - fix(user.ip.restriction): prevent removing last ip on deny default rule

Signed-off-by: Antoine Leblanc <antoine.leblanc@ovhcloud.com>

### CODE REVIEW

**Review by lizardK — DISMISSED** [review]

**Review by cbourgois — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/account/user/translations/Messages_en_GB.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
   "user_ipRestrictions_title": "Access restricted by IP",
   "user_ipRestrictions_activate": "Enable",
   "user_ipRestrictions_default_rule": "Default rule",
+  "user_ipRestrictions_warning_last_ip_with_deny": "Please change the default rule to «Authorized» in order to be able to remove the latest IP address.",
```
Comment (Alkorin) [inline packages/manager/apps/dedicated/client/app/account/user/translations/Messages_en_GB.json:69]:
```suggestion
  "user_ipRestrictions_warning_last_ip_with_deny": "Please change the default rule to «Authorized» in order to be able to remove the last IP address.",
```

**Review by Alkorin — CHANGES_REQUESTED** [review]
Doesn't work as intended, the banner is always displayed when defaultRule is allowed

File: packages/manager/apps/dedicated/client/app/account/user/security/user-security.html [hunk 2]
```diff
@@ -776,6 +776,16 @@ <h5 class="media-heading">
 
                             <hr />
 
+                            <oui-message
+                                class="mb-3"
+                                type="warning"
+                                ng-show="!canRemoveIp()"
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/account/user/security/user-security.html:782]:
```suggestion
                                ng-if="!canRemoveIp()"
```

File: packages/manager/apps/dedicated/client/app/account/user/ip/restriction/user-ip-restriction.controller.js [hunk 3]
```diff
@@ -156,5 +156,9 @@ angular
         $scope.ipRestrictionCurrentEdit = null;
         $scope.ipRestrictionCurrentEditBack = null;
       };
+
+      $scope.canRemoveIp = function canRemoveIp() {
+        return $scope.defaultRule.rule === 'deny' && $scope.list.length > 1;
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/account/user/ip/restriction/user-ip-restriction.controller.js:161]: shouldn't it be 
```suggestion
        return $scope.defaultRule.rule !== 'deny' || $scope.list.length > 1;
```
? (according to the content added )

Comment (antleblanc) [reply packages/manager/apps/dedicated/client/app/account/user/ip/restriction/user-ip-restriction.controller.js]: Yes! I've replaced `!== 'deny'` to `=== 'accept'`.

**Review by marie-j — DISMISSED** [review]

**Review by ganeshkumar1989 — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/account/user/ip/restriction/user-ip-restriction.controller.js [hunk 4]
```diff
@@ -54,8 +52,8 @@ angular
           .then(
             (defaultRule) => {
               $scope.defaultRule = defaultRule;
+              $scope.ipRestrictionDefaultRule = angular.copy(defaultRule);
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/account/user/ip/restriction/user-ip-restriction.controller.js:55]: Can't we avoid using `angular.copy` ?

**Review by JeremyDec — DISMISSED** [review]

**Comment by antleblanc** [comment]
Rebasing on `develop` branch (got 1 approval)

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Review by JeremyDec — DISMISSED** [review]

**Review by marie-j — APPROVED** [review]

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Review by marie-j — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**